uni/lib/view/common_widgets/pages_layouts/secondary/secondary.dart

```diff
@@ -1,21 +1,42 @@
+import 'package:flutter/foundation.dart';
 import 'package:flutter/material.dart';
 import 'package:uni/view/common_widgets/pages_layouts/general/general.dart';
-import 'package:uni/view/common_widgets/pages_layouts/general/widgets/app_bar.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/widgets/bottom_navigation_bar.dart';
 import 'package:uni/view/common_widgets/pages_layouts/general/widgets/refresh_state.dart';
+import 'package:uni/view/common_widgets/pages_layouts/general/widgets/top_navigation_bar.dart';
 
 /// Page with a back button on top
 abstract class SecondaryPageViewState<T extends StatefulWidget>
     extends GeneralPageViewState<T> {
   @override
   Scaffold getScaffold(BuildContext context, Widget body) {
     return Scaffold(
-      appBar: CustomAppBar(getTopRightButton: getTopRightButton),
+      appBar: getTopNavbar(context),
+      bottomNavigationBar: AppBottomNavbar(
+        parentContext: context,
+      ),
       body: RefreshState(onRefresh: onRefresh, child: body),
     );
   }
 
+  String? getTitle() {
+    return null;
+  }
+
+  Widget? getTopRightButton(BuildContext context) {
+    return null;
+  }
+
   @override
-  Widget getTopRightButton(BuildContext context) {
-    return Container();
+  @nonVirtual
+  AppTopNavbar? getTopNavbar(BuildContext context) {
+    return AppTopNavbar(
+      title: getTitle(),
+      leftButton: const Padding(
+        padding: EdgeInsets.symmetric(horizontal: 5),
+        child: BackButton(),
```

Comment:
This "back button" is not working well when navigating via the bottom bar buttons. For example, if I am in the home page and press the restaurants button, then press the back button, nothing happens, but we should go back to the home page.